cmd/gas/main.go

```diff
@@ -0,0 +1,232 @@
+
+	logger *log.Logger
+)
+
+// #nosec
+func usage() {
+
+	fmt.Fprintln(os.Stderr, usageText)
+	fmt.Fprint(os.Stderr, "OPTIONS:\n\n")
+	flag.PrintDefaults()
+	fmt.Fprint(os.Stderr, "\n\nRULES:\n\n")
+
+	// sorted rule list for eas of reading
+	rl := rules.Generate()
+	keys := make([]string, 0, len(rl))
+	for key := range rl {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+	for _, k := range keys {
+		v := rl[k]
+		fmt.Fprintf(os.Stderr, "\t%s: %s\n", k, v.Description)
+	}
+	fmt.Fprint(os.Stderr, "\n")
+}
+
+func loadConfig(configFile string) (gas.Config, error) {
+	config := gas.NewConfig()
+	if configFile != "" {
+		file, err := os.Open(configFile)
+		if err != nil {
+			return nil, err
+		}
+		defer file.Close()
+		if _, err := config.ReadFrom(file); err != nil {
+			return nil, err
+		}
+	}
+	if *flagIgnoreNoSec {
+		config.SetGlobal("nosec", "true")
+	}
+	return config, nil
+}
+
+func loadRules(include, exclude string) rules.RuleList {
+	filters := make([]rules.RuleFilter, 0)
+	if include != "" {
+		including := strings.Split(include, ",")
+		filters = append(filters, rules.NewRuleFilter(false, including...))
+	}
+
+	if exclude != "" {
+		excluding := strings.Split(exclude, ",")
+		filters = append(filters, rules.NewRuleFilter(true, excluding...))
+	}
+	return rules.Generate(filters...)
+}
+
+func saveOutput(filename, format string, issues []*gas.Issue, metrics *gas.Metrics) error {
+	if filename != "" {
+		outfile, err := os.Create(filename)
+		if err != nil {
+			return err
+		}
+		defer outfile.Close()
+		output.CreateReport(outfile, format, issues, metrics)
+	} else {
+		output.CreateReport(os.Stdout, format, issues, metrics)
+	}
+	return nil
+}
+
+func main() {
+
+	// Setup usage description
+	flag.Usage = usage
+
+	// Parse command line arguments
+	flag.Parse()
+
+	// Ensure at least one file was specified
+	if flag.NArg() == 0 {
+		fmt.Fprintf(os.Stderr, "\nError: FILE [FILE...] or './...' expected\n")
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	// Setup logging
+	logWriter := os.Stderr
+	if *flagLogfile != "" {
+		var e error
+		logWriter, e = os.Create(*flagLogfile)
+		if e != nil {
+			flag.Usage()
+			log.Fatal(e)
+		}
+	}
+	logger = log.New(logWriter, "[gas] ", log.LstdFlags)
+
+	// Load config
+	config, err := loadConfig(*flagConfig)
+	if err != nil {
+		logger.Fatal(err)
+	}
+
+	// Load enabled rule definitions
+	ruleDefinitions := loadRules(*flagRulesInclude, *flagRulesExclude)
+
+	// Create the analyzer
+	analyzer := gas.NewAnalyzer(config, logger)
+	analyzer.LoadRules(ruleDefinitions.Builders()...)
+
+	vendor := regexp.MustCompile(`[\\/]vendor([\\/]|$)`)
+
+	// Iterate over packages on the import paths
+	for _, pkg := range gotool.ImportPaths(flag.Args()) {
+
+		// Skip vendor directory
+		if vendor.MatchString(pkg) {
+			continue
+		}
+
+		abspath, _ := filepath.Abs(pkg)
+		logger.Println("Searching directory:", abspath)
+		if err := analyzer.Process(pkg); err != nil {
+			logger.Fatal(err)
```

Comment:
Maybe it is worth continuing the processing if only one package fails.

Reply:
I think the causes of errors here are: 

1. It is a valid go package but it doesn't compile
2. It is not a valid go package

I think I was trying to mimic the compiler behaviour here. For the purposes of a mass scan I 
could see how this would be useful to continue processing. 

However I would typically do something like to scan everything in my workspace: 

```bash
# (mac osx)
$ cd $GOPATH/src && go list ./... | xargs  -I {} gas $GOPATH/src/{}/...
```

Reply:
Agreed. The workaround is fair enough.